game.py

```diff
@@ -0,0 +1,359 @@
+    skill_effects = {
+       
+        "Shield Bash": lambda: shield_bash(hero, enemies),
+        "Swift Slash": lambda: swift_slash(hero, enemies),
+        "Lance Charge": lambda: lance_charge(hero, enemies),
+        "Longshot": lambda: longshot(hero, enemies),
+        # Add more skills here if needed
+    }
+
+    # Execute the corresponding skill
+    skill_effects.get(skill_name, lambda: print_with_delay(f"{hero.name} This skill is unlock. No action taken."))()
+
+def shield_bash(hero, enemies):
+    skill_damage = hero.attack * 1.5  # Example damage multiplier for Shield Bash
+    target = enemies[0]  # Targeting the first enemy for simplicity
+    target.health -= skill_damage
+    print_with_delay(f"{hero.name} used Shield Bash on {target.name} for {skill_damage} damage!")
+    check_enemy_status(hero, target, enemies)
+
+def swift_slash(hero, enemies):
+    skill_damage = hero.attack * 2  # Example damage multiplier for Swift Slash
+    target = enemies[0]
+    target.health -= skill_damage
+    print_with_delay(f"{hero.name} used Swift Slash on {target.name} for {skill_damage} damage!")
+    check_enemy_status(hero, target, enemies)
+
+def lance_charge(hero, enemies):
+    skill_damage = hero.attack * 1.8  # Example multiplier for Lance Charge
+    target = enemies[0]
+    target.health -= skill_damage
+    print_with_delay(f"{hero.name} used Lance Charge on {target.name} for {skill_damage} damage!")
+    check_enemy_status(hero, target, enemies)
+
+def longshot(hero, enemies):
+    skill_damage = hero.attack * 1.3  # Example multiplier for Longshot
+    target = enemies[0]
+    target.health -= skill_damage
+    print_with_delay(f"{hero.name} used Longshot on {target.name} for {skill_damage} damage!")
+    check_enemy_status(hero, target, enemies)
+
+def check_enemy_status(hero, target, enemies):
+    if target.health <= 0:
+        print_with_delay(f"\nThe {target.name} has been defeated!")
+        hero.xp += 5  # Adds 5 XP
+        hero.check_level_up()  # Check if the hero should level up
+        hero.earn_coins(random.randint(5, 10))  # Hero earns random coins between 5 and 10
+        enemies.remove(target)  # Remove defeated enemy
+    else:
+        print_with_delay(f"{target.name}'s Health: {target.health}")
+
+def attack_action(hero, enemies):
+    target = enemies[0]  # For simplicity, just attacking the first enemy
+    damage = max(1, hero.attack - target.defense)
+    target.health -= damage
+    print_with_delay(f"You attacked the {target.name} for {damage} damage!")
+
+    if target.health <= 0:
+        print_with_delay(f"\nThe {target.name} has been defeated!")
+        hero.xp +=(5)
+        hero.earn_coins(random.randint(5, 10))
+        enemies.remove(target)
+    else:
+        print_with_delay(f"\n{target.name}'s health: {target.health}")
+
+def defend_action(hero):
+    print_with_delay(f"{hero} prepare to block an attack, reducing damage next turn.")
```

Comment:
Forgot to call the name
```suggestion
    print_with_delay(f"{hero.name} prepare to block an attack, reducing damage next turn.")
```